pkg/graveler/committed/merge_iterator.go

```diff
@@ -192,11 +125,113 @@ func (d *compareIterator) Step() bool {
 	return false
 }
 
+// stepValue moves one step according to current value
+// returns hasMore if iterator has more, and done if the step is over  (got to a value, end of iterator, or error)
+func (d *compareIterator) stepValue() (bool, bool) {
+	val, rngDiff := d.diffIt.Value()
+	key := val.Key
+	typ := val.Type
+	baseVal, err := d.valueFromBase(key)
+	if err != nil {
+		d.err = err
+		return false, true
+	}
+	switch typ {
+	case graveler.DiffTypeAdded:
+		// exists on source, but not on dest
+		if baseVal == nil {
+			// added only on source
+			d.val = val.Copy()
+			d.setRangeDiff(rngDiff)
```

Comment:
The iterators return range when they can, so that if you are Iterating through a range even if you get a value you could get the current range and understand what to expect on calling nextRange. 
The reason they do not always return range is that they don't always have a range (e.g diff of two overlapping ranges)

Reply:
It's the same behavior as `committed.iterator`, with the exception that we don't always have a range